lib/ansible/plugins/strategy/__init__.py

```diff
@@ -1130,21 +1130,20 @@ def _evaluate_conditional(h):
 
         skipped = False
         msg = ''
+        self._tqm.send_callback('v2_playbook_on_task_start', task, is_conditional=False)
+
+        # These don't support "when" conditionals
+        if meta_action in ('noop', 'flush_handlers', 'refresh_inventory', 'reset_connection') and task.when:
```

Comment:
This is potentially a small change in behavior since one of the conditions below is `elif meta_action == 'refresh_inventory' or self.flush_cache:`. I think that's fine though, because as I'm digging into that code again I think the `or self.flush_cache` is wrong and I'm going to work on fixing that (it shouldn't be in the strategy at all).

Reply:
This is a good catch, not sure why I missed that. I could copy the `or self.flush_cache`, or just leave it as-is, or forego this little "opportunistic" de-dupe for now. Fine with any of those options.